README.md

```diff
@@ -1,52 +1,90 @@
+From these, all other units can be derived. 
 
-All basic units and many combined units are predefined in the library, but it is easy to define your own units. For example, if you want to measure the weight of cables, you might need a unit mass/length:
+(Note that this means that unlib does not provide units like, e.g., _kilobytes_. While it is easy to add other units to this library's system of units, unlib was specifically not designed to be a framework for _any_ unit system. If you need to create your own type systems, you might want to look into [Boost.Units](https://www.boost.org/doc/libs/1_75_0/doc/html/boost_units.html). 
+
+## Dimensions
+
+The basic units, and all units which can be derived from them, represent physical dimensions. They are described by the exponents of the underlying seven basic units. A dimension, in the sense of this library, therefore is a (compile-time) set of the seven exponents, representing the seven basic units. For example, electrical charge, measured in Coulomb, a.k.a. Ampere second, can be defined this way:
+
+```cpp
+using my_dimension = unlib::dimension< unlib::ratio_t<1>  
+                                     , unlib::ratio_t<0> 
+                                     , unlib::ratio_t<0> 
+                                     , unlib::ratio_t<1> 
+                                     , unlib::ratio_t<0> 
+                                     , unlib::ratio_t<0> 
+                                     , unlib::ratio_t<0> >;
+```
+
+Since the library's way of telling these exponents from each other is their order, they must always be passed in the correct order, which is an invitation to silly errors. As a human, in order to even stand a chance to write this without making an error, you would have to write it this way:
+
+```cpp
+using my_dimension = unlib::dimension< /* time               */ unlib::ratio_t<1>  
+                                     , /* mass               */ unlib::ratio_t<0> 
+                                     , /* length             */ unlib::ratio_t<0> 
+                                     , /* current            */ unlib::ratio_t<1> 
+                                     , /* temperature        */ unlib::ratio_t<0> 
+                                     , /* substance amount   */ unlib::ratio_t<0> 
+                                     , /* luminous intensity */ unlib::ratio_t<0> >;
+```
+
+However, dimensions can, and should, be built in different, less error-prone, ways. For example, the `dimension_t` meta function creates dimensions from the exponents of the seven basic units, passed in any order. Using it, electrical charge can be defined this way:
 
 ```c++
-using cable_mass = unlib::div_unit_t< unlib::unlib::mass, unlib::length >
+using electrical_charge = unlib::dimension_t<unlib::current, unlib::time>;
```

Comment:
This would be much clearer written as
`using electrical_charge = unlib::mul_dimension_t<unlib::current, unlib::time>;`

IMHO, the "promoted" way to derive dimensions should be in terms of operations, not using `unlib::dimension_t` directly, as in this example. I would even go so far as to not expose this as user interface. This would also open up the opportunity to simplify the code by removing the metafunctions to handle "unsorted unit lists".

Reply:
OK. I will look into this and maybe remove `dimension_t` from the interface altogether.